### ISSUE TITLE
`cargo --help` panics when stdout is closed

### ISSUE THREAD

**Issue by lopopolo** [rust-lang/cargo#10131]
### Problem

`cargo --help` panics when stdout is closed.

There is a discussion about this API papercut upstream: https://github.com/clap-rs/clap/discussions/3055.

### Steps

```console
$ RUST_BACKTRACE=1 cargo --help | false
thread 'main' panicked at 'Error writing Error to stdout: Os { code: 32, kind: BrokenPipe, message: "Broken pipe" }', /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-2.33.3/src/errors.rs:401:55
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: core::result::Result<T,E>::expect
   4: clap::errors::Error::exit
   5: cargo::exit_with_error
   6: cargo::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

### Possible Solution(s)

Use the `App::get_matches_from_safe` API and manually handle the error it returns, suppressing failures to write to either stdout or stderr.

### Version

```text
cargo 1.56.0 (4ed5d137b 2021-10-04)
release: 1.56.0
commit-hash: 4ed5d137baff5eccf1bae5a7b2ae4b57efad4a7d
commit-date: 2021-10-04
```

**Comment by weihanglo** [comment]
I saw it fixed on clap v2.33.4. Really great!
https://github.com/clap-rs/clap/commit/accf3d251168e10ecbb0b4ae200c3196b950dac9

**Comment by ehuss** [comment]
Thanks @weihanglo, would you mind posting a PR that updates our `Cargo.toml` to "2.34.0"?
